drivers/bmp180/bmp180.c

```diff
@@ -65,10 +65,12 @@ int bmp180_init(bmp180_t *dev, i2c_t i2c, uint8_t mode)
     /* Acquire exclusive access */
     i2c_acquire(dev->i2c_dev);
 
-    /* Check sensor ID */
-    char checkid;
-    i2c_write_byte(dev->i2c_dev, BMP180_ADDR, BMP180_REGISTER_ID);
-    i2c_read_reg(dev->i2c_dev, BMP180_ADDR, BMP180_REGISTER_ID, &checkid);
+    /* Check sensor ID (retry a few times, it may fail at first) */
+    char checkid = 0;
+    for (size_t ix = 0; checkid == 0 && ix < 3; ++ix) {
+	i2c_write_byte(dev->i2c_dev, BMP180_ADDR, BMP180_REGISTER_ID);
```

Comment:
Again.
The above mentioned i2c_write_byte still does not make sense. Why is it needed? I don't think it is. But if I leave it out I'm getting all zeroes for the calibration.

To solve that I need a delay between the BMP180_REGISTER_ID and the BMP180_CALIBRATION_AC1.

Here is a question to you. Can you try the following?

    diff --git a/drivers/bmp180/bmp180.c b/drivers/bmp180/bmp180.c
    index e772c3a..ad771cc 100644
    --- a/drivers/bmp180/bmp180.c
    +++ b/drivers/bmp180/bmp180.c
    @@ -73,10 +73,10 @@ int bmp180_init(bmp180_t *dev, i2c_t i2c, uint8_t mode)
             i2c_release(dev->i2c_dev);
             return -1;
         }
    +    xtimer_usleep(3);
 
         char buffer[22] = {0};
         /* Read calibration values, using contiguous register addresses */
    -    i2c_write_byte(dev->i2c_dev, BMP180_ADDR, (char)BMP180_CALIBRATION_AC1);
         if (i2c_read_regs(dev->i2c_dev, BMP180_ADDR, BMP180_CALIBRATION_AC1, buffer, 22) < 0) {
             DEBUG("[Error] Cannot read calibration registers.\n");
             i2c_release(dev->i2c_dev);